src/ui/session/content/message_row/reply.rs

```diff
@@ -97,19 +97,28 @@ impl MessageReply {
         let imp = self.imp();
 
         let message = self.message().unwrap();
-        let reply_to_message_id = message.reply_to_message_id();
         let is_outgoing = message.is_outgoing();
-        let chat = if message.reply_in_chat_id() != 0 {
-            message.chat_().session_().chat(message.reply_in_chat_id())
-        } else {
-            message.chat_()
+        let reply_to = message.reply_to().unwrap().0;
+
+        match reply_to {
+            tdlib::enums::MessageReplyTo::Message(reply_to) => {
+                let reply_to_message_id = reply_to.message_id;
+                let chat = if reply_to.chat_id != 0 {
+                    message.chat_().session_().chat(reply_to.chat_id)
+                } else {
+                    message.chat_()
+                };
+
+                match chat.fetch_message(reply_to_message_id).await {
+                    Ok(message) => self.update_from_message(&message, is_outgoing),
+                    Err(_) => imp.message_label.set_label("Deleted message"),
+                };
+            }
+            tdlib::enums::MessageReplyTo::Story(_) => {
+                // TODO: Implement story replies
+                unimplemented!()
```

Comment:
Is this match arm really unreachable? Otherwise, this is going to crash the app.

Reply:
Yes, it's blocked here:
https://github.com/alissonlauffer/paper-plane/blob/tdlib-1.8.18/src/ui/session/content/message_row/bubble.rs#L181-L186
and even it crashed here, it wouldn't crash the whole app, and just display an infinite "Loading..." reply.